session/session.go

```diff
@@ -2843,30 +2844,77 @@ func logStmt(execStmt *executor.ExecStmt, s *session) {
 	}
 }
 
-func logGeneralQuery(execStmt *executor.ExecStmt, s *session, isPrepared bool) {
+func genFnGetQuery(execStmt *executor.ExecStmt, s *session, isPrepared bool) func(buf *strings.Builder) string {
 	vars := s.GetSessionVars()
-	if variable.ProcessGeneralLog.Load() && !vars.InRestrictedSQL {
-		var query string
+	return func(buf *strings.Builder) string {
 		if isPrepared {
-			query = execStmt.OriginText()
+			buf.WriteString(execStmt.OriginText())
 		} else {
-			query = execStmt.GetTextToLog()
+			buf.WriteString(execStmt.GetTextToLog())
 		}
 
-		query = executor.QueryReplacer.Replace(query)
+		bufString := buf.String()
+		queryMutable := *(*[]byte)(unsafe.Pointer(&bufString))
+		for i, b := range queryMutable {
+			if b == '\r' || b == '\n' || b == '\t' {
+				queryMutable[i] = ' '
+			}
+		}
```

Comment:
Maybe we no longer need this, as we are very sure that the SQL of the general log can be correctly encoded?

Reply:
LGTM, this is just for compatibility of legacy code, WDYT @crazycs520